core/logger/logger.go

```diff
@@ -163,16 +192,132 @@ func (l *zapLogger) withCallerSkip(skip int) Logger {
 
 func (l *zapLogger) ErrorIf(err error, msg string) {
 	if err != nil {
+		sentry.CaptureException(err)
 		l.withCallerSkip(1).Errorw(msg, "err", err)
 	}
 }
 
 func (l *zapLogger) ErrorIfClosing(c io.Closer, name string) {
 	if err := c.Close(); err != nil {
+		sentry.CaptureException(err)
 		l.withCallerSkip(1).Errorw(fmt.Sprintf("Error closing %s", name), "err", err)
 	}
 }
 
+func (l *zapLogger) Error(args ...interface{}) {
+	hub := sentry.CurrentHub().Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetContext("logger", map[string]interface{}{
+			"args": args,
+		})
+		scope.SetLevel(sentry.LevelError)
+	})
+	hub.CaptureMessage(fmt.Sprintf("%v", args))
+	l.SugaredLogger.Error(args...)
```

Comment:
With the extra level of indirection, these need to skip another call level for the stack trace to point to the right spot.
Can drop in `.withCallerSkip(1).` like `ErrorIf*` for a quick fix, but if this is no longer an exception then we should probably be caching it on the zapLogger.

Reply:
Alternatively, we could try to hook in to zap's output rather than trying to intercept on the way in.

Reply:
https://github.com/TheZeroSlave/zapsentry

Reply:
I don't really want to pull in another dependency for this, especially something small which we could easily write ourselves.

It is not possible to use zap hooks here because they don't expose enough information (e.g. extra log fields).